### PR TITLE
Add DataStream support

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

File: protobufs/livekit_models.proto [hunk 1]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
...
+  
+    // oneof to choose between specific header types
+    oneof content_header {
+      TextHeader textHeader = 9;
+      FileHeader fileHeader = 10;
+    }
+  }
+
+  message Packet {
+    string messageId = 1;  // unique identifier for this data stream to map it to the correct header
```
Comment (boks1971) [inline protobufs/livekit_models.proto:648]: not calling this `streamId`?

File: protobufs/livekit_models.proto [hunk 2]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
...
+    uint64 chunkIndex = 2; // chunkId as an unsigned 32-bit integer
```
Comment (boks1971) [inline protobufs/livekit_models.proto:649]: comment says 32-bit, type here is 64-bit. Guessing 32-bit should be enough for this?

File: protobufs/livekit_models.proto [hunk 3]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
...
+    bytes content = 3; // content as binary (bytes)
+    uint32 contentLength = 4; // content length in bytes
```
Comment (boks1971) [inline protobufs/livekit_models.proto:651]: what is the purpose of this? can this not be derived from `content` as length of it? Are there cases where content is padded?  (maybe for the encryption case?)

File: protobufs/livekit_models.proto (hunk 2 above)
Comment (boks1971) [reply protobufs/livekit_models.proto:649]: oh, looks like you are making both length and chunkId 64-bit, comment needs update then.

File: protobufs/livekit_models.proto [hunk 4]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
   TF_ENHANCED_NOISE_CANCELLATION = 5;
 }
 
+message DataStream {
+
+  // enum for operation types (specific to TextStreamHeader)
```
Comment (boks1971) [inline protobufs/livekit_models.proto:606]: :nit: there is no message called `TextStreamHeader`. When it is written like that, it makes it look like a type and I would want to search for it to look at the shape of it.

File: protobufs/livekit_models.proto [hunk 5]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
...
+    bool generated = 5; // true if the text has been generated by an agent from a participant's audio transcription
+
+  }
+  
+  // header properties specific to file or image streams
+  message FileHeader {
+    string fileName = 1;  // optional: name of the file
+  }
+
+  // main DataStreamHeader that contains a oneof for specific headers
```
Comment (boks1971) [inline protobufs/livekit_models.proto:629]: :nit: maybe change comment to say `DataStream.Header` to signify that it is a not a type of its own.

File: protobufs/livekit_models.proto [hunk 6]
```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
...
+    int32 version = 2;  // Optional: Version for updates/edits
+    string replyToStreamId = 3;  // Optional: Reply to specific message
+    repeated string attachedStreamIds = 4;
+    bool generated = 5; // true if the text has been generated by an agent from a participant's audio transcription
+
+  }
+  
+  // header properties specific to file or image streams
+  message FileHeader {
+    string fileName = 1;  // optional: name of the file
```
Comment (boks1971) [inline protobufs/livekit_models.proto:626]: It says optional, but shouldn't this be mandatory for type file?

**Review by boks1971 — APPROVED** [review]

File: protobufs/livekit_models.proto (hunk 3 above)
Comment (lukasIO) [reply protobufs/livekit_models.proto:651]: yeah, although I'm not sure anymore what we'd need it for, even in the encryption case. Will remove and we can re-add if it turns out we need it

File: protobufs/livekit_models.proto (hunk 2 above)
Comment (lukasIO) [reply protobufs/livekit_models.proto]: done!

File: protobufs/livekit_models.proto (hunk 1 above)
Comment (lukasIO) [reply protobufs/livekit_models.proto]: thanks for catching this, was an oversight

**Review by cnderrauber — APPROVED** [review]

**Review by theomonnom — APPROVED** [review]

**Review by nbsp — APPROVED** [review]

**Pull request merged**